### PR TITLE
Updated to use UnparsedResourceName

### PR DESCRIPTION
**Pull request by chrisdunelm**

### CODE REVIEW

File: Google.Api.Gax/UnparsedResourceName.cs [hunk 1]
```diff
@@ -12,30 +12,30 @@ namespace Google.Api.Gax
     /// <summary>
     /// A resource name in which nothing is known about the name structure.
     /// </summary>
-    public sealed class UnknownResourceName : IResourceName, IEquatable<UnknownResourceName>
+    public sealed class UnparsedResourceName : IResourceName, IEquatable<UnparsedResourceName>
     {
         /// <summary>
-        /// Parse a resource name into an <see cref="UnknownResourceName"/>.
+        /// Parse a resource name into an <see cref="UnparsedResourceName"/>.
```
Comment (jskeet) [inline Google.Api.Gax/UnparsedResourceName.cs]: This sentence is kinda funky - "We're parsing it but we're not really." I think it's okay though.

File: Google.Api.Gax/UnparsedResourceName.cs [hunk 2]
```diff
@@ -12,30 +12,30 @@ namespace Google.Api.Gax
...
         /// Only minimal verification is carried out that <paramref name="name"/> is a value resource name string.
```
Comment (jskeet) [inline Google.Api.Gax/UnparsedResourceName.cs]: Nit: value => valid, while we're editing this file?

Comment (chrisdunelm) [reply Google.Api.Gax/UnparsedResourceName.cs]: done

File: Google.Api.Gax/UnparsedResourceName.cs (hunk 1 above)
Comment (chrisdunelm) [reply Google.Api.Gax/UnparsedResourceName.cs]: ack

**Pull request merged**